common/src/main/java/com/linecorp/centraldogma/common/RevisionRange.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2018 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.common;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * A class which contains {@link Revision}s of {@code from} and {@code to}.
+ */
+public class RevisionRange {
+
+    private final Revision from;
+
+    private final Revision to;
+
+    /**
+     * Creates a new instance with the specified integer value of {@code from} and {@code to}.
+     */
+    public RevisionRange(int from, int to) {
+        this(new Revision(from), new Revision(to));
+    }
+
+    /**
+     * Creates a new instance with the specified {@code from} {@link Revision} and {@code to} {@link Revision}.
+     */
+    public RevisionRange(Revision from, Revision to) {
+        this.from = requireNonNull(from, "from");
+        this.to = requireNonNull(to, "to");
+    }
+
+    public Revision from() {
+        return from;
+    }
+
+    public Revision to() {
+        return to;
+    }
+
+    /**
+     * Returns a {@link RevisionRange} whose major value of {@code from} {@link Revision} is lower than
+     * or equal to the major value of {@code to} {@link Revision}.
+     *
+     * @throws UnsupportedOperationException if {@link #isRelative()} returns {@code true}
+     */
+    public RevisionRange toAscending() {
+        if (isAscending() || from.equals(to)) {
+            return this;
+        }
+
+        return new RevisionRange(to, from);
+    }
+
+    /**
+     * Returns a {@link RevisionRange} whose major value of {@code from} {@link Revision} is greater than
+     * or equal to the major value of {@code to} {@link Revision}.
+     *
+     * @throws UnsupportedOperationException if {@link #isRelative()} returns {@code true}
+     */
+    public RevisionRange toDescending() {
+        if (isAscending()) {
+            return new RevisionRange(to, from);
+        }
+
+        return this;
+    }
+
+    /**
+     * Returns {@code true} if the major value of {@code from} {@link Revision} is lower than the major
+     * value of {@code to} {@link Revision}.
+     *
+     * @throws UnsupportedOperationException if {@link #isRelative()} returns {@code true}
+     */
+    public boolean isAscending() {
```

Comment:
nit: Just curious. How about returning `true` or `false` here instead of throwing an exception? Then, throw the exception when this range is relative in `toAscending()` and `toDescending`? (adding some private method like `ensureAbsoluteRange()`)

Reply:
I think it can confuse the user when it returns `false`. If it returns `false`, a user might think that this is descending or equal revision. WDYT @hyangtack?